compiler/rustc_codegen_llvm/src/abi.rs

```diff
@@ -141,7 +141,7 @@ impl LlvmType for CastTarget {
 
         if self.prefix.iter().all(|x| x.is_none()) {
             // Simplify to a single unit when there is no prefix and size <= unit size
-            if self.rest.total <= self.rest.unit.size {
+            if self.rest.total <= self.rest.unit.size && !self.rest.force_array {
```

Comment:
Could you please add an assertion that force_array is not true at https://github.com/rust-lang/rust/blob/e3df96cfda905301fc8514e000f942222c1ab6ad/compiler/rustc_codegen_cranelift/src/abi/pass_mode.rs#L42 Cranelift IR doesn't jave a way to represent array types.

Reply:
After having looked into this more, I suspect that the cranelift lowering actually is not quite correct and you'll have to handle the new is_consecutive flag somehow. Or at least, I don't see how your current code would ensure consecutive allocation. Note that this part applies to not just ppc64le, but also to aarch64.

Reply:
abi-cafe doesn't show any issues on aarch64 last time I checked.